lib/brainstem/api_docs/formatters/open_api_specification/version_2/presenter_formatter.rb

```diff
@@ -52,18 +52,61 @@ def format_type!
             end
 
             def format_fields!
-              return unless presenter.valid_fields.any?
+              return unless presenter.valid_fields.any? || presenter.valid_associations.any?
 
-              definition.merge! properties: format_field_branch(presenter.valid_fields)
+              properties = format_field_branch(presenter.valid_fields)
+              with_associations = format_field_associations(properties)
+
+              definition.merge! properties: with_associations
             end
 
             def format_field_branch(branch)
-              branch.inject(ActiveSupport::HashWithIndifferentAccess.new) do |buffer, (name, field)|
+              branch.each_with_object(ActiveSupport::HashWithIndifferentAccess.new) do |(name, field), buffer|
                 buffer[name.to_s] = format_field(field)
-                buffer
               end
             end
-            
+
+            def format_field_associations(properties)
+              presenter.valid_associations.each_with_object(properties) do |(name, association), props|
+                if association.polymorphic?
+                  key = association.name + "_ref"
+                  props[key] = {
+                    type: 'object',
+                    description: association_description(key, name),
+                    properties: {
+                      key: type_and_format(:string),
+                      id: type_and_format(:string)
+                    }
+                  }
+                  next
+                end
+
+                key = association_key(association)
+                description = association_description(key, association.name)
+                formatted_type = if association.type == :has_many
+                  type_and_format(:array, :integer)
+                else
+                  type_and_format(:integer)
+                end.merge(description: description)
+
+                props[key] = formatted_type unless props[key]
+              end
+            end
+
+            def association_description(key, name)
+              "`#{key}` will only be included in the response if `#{name}` is in the list of included associations."
+            end
+
+            def association_key(association)
+              key = if association.foreign_key
+                association.foreign_key
+              else
+                "#{association.name.singularize}_id"
+              end
+
+              association.type == :has_many ? "#{key}s" : key
```

Comment:
Shouldn't this belong on line 104? I feel its weird to pluralize the foreign key given by the user.

Reply:
with a `:has_many`, the user wouldn't be providing a foreign key, this `key` at this point is generated by the `else` block via ` "#{association.name.singularize}_id"`

Reply:
👍